src/app/pages/send-tokens/hooks/use-selected-asset-balance.ts

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from 'react';
+
+import { getTicker } from '@app/common/utils';
+import { ftDecimals, stacksValue } from '@app/common/stacks-utils';
+import { useSelectedStacksCryptoAssetBalance } from '@app/query/stacks/balance/crypto-asset-balances.hooks';
+
+export function useSelectedAssetBalance(assetId: string) {
+  const selectedAssetBalance = useSelectedStacksCryptoAssetBalance(assetId);
+
+  return useMemo(() => {
+    const isStx =
+      selectedAssetBalance?.asset.blockchain === 'stacks' &&
+      selectedAssetBalance.asset.type === 'crypto-currency';
+
+    const stxBalance = stacksValue({
+      value: selectedAssetBalance?.balance.amount || 0,
+      withTicker: false,
+    });
+
+    const ftBalance = selectedAssetBalance?.asset.decimals
+      ? ftDecimals(selectedAssetBalance.balance.amount, selectedAssetBalance.asset.decimals)
+      : selectedAssetBalance?.balance.amount.toFormat();
+    const ticker = selectedAssetBalance
+      ? selectedAssetBalance.asset.symbol || getTicker(selectedAssetBalance.asset.name)
+      : null;
+    const hasDecimals =
+      selectedAssetBalance?.asset.decimals && selectedAssetBalance.asset.decimals > 0;
+    const placeholder = `0${
+      hasDecimals ? `.${'0'.repeat(isStx ? 6 : selectedAssetBalance.asset.decimals || 0)}` : ''
+    } ${ticker}`;
+
+    return {
+      balance: stxBalance || ftBalance,
+      balanceBigNumber: selectedAssetBalance?.balance.amount,
```

Comment:
doesn't `balanceBigNumber === balance.amount`?

Reply:
Good point, this was here but not sure why it is needed. I'll check and remove.